lucene/core/src/java/org/apache/lucene/geo/XYRectangle.java

```diff
@@ -29,12 +31,16 @@
 
   /** Constructs a bounding box by first validating the provided x and y coordinates */
   public XYRectangle(double minX, double maxX, double minY, double maxY) {
-    this.minX = minX;
-    this.maxX = maxX;
-    this.minY = minY;
-    this.maxY = maxY;
-    assert minX <= maxX;
-    assert minY <= maxY;
+    if (minX > maxX) {
```

Comment:
I wonder if a XYRectangle should be initialise with floats instead of doubles like the other XYGeometries?